### PR TITLE
Replace stored procedure

### PR DESCRIPTION
**Pull request by tej99**
# Motivation and Context

Currently a stored procedure is being used to create actions off the back of actionPlanJobs. To give more visibility of what's going on, this function is being shifted from the database layer into the application. 

# What has changed

No more stored procedure. The behaviour is the same just in the JPA repository instead. 

# How to test?

Validate tests pass.

# Links

https://trello.com/c/bj6mBW57/129-remove-stored-procedure-to-create-actions-in-action-service

### CODE REVIEW

**Comment by benjefferies** [comment]
👏

**Comment by tej99** [comment]
[Remove stored procedure to create actions in action-service](https://trello.com/c/bj6mBW57/129-remove-stored-procedure-to-create-actions-in-action-service)

File: src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionServiceImpl.java [hunk 1]
```diff
@@ -169,6 +183,35 @@ public Action createAction(final Action action) {
...
+        actionPlanRepository.findByActionPlanPK(actionPlanJob.getActionPlanFK());
+
+    Timestamp currentTime = new Timestamp((new Date()).getTime());
+
+    if (actionCaseRepository.hasActiveCaseWithActionPlanId(
+        actionPlan.getActionPlanPK(), currentTime)) {
+      actionRepo.createActionsForRulesDueAtTime(actionPlan.getActionPlanPK(), currentTime);
+    }
+
+    actionPlanJob.setUpdatedDateTime(currentTime);
```
Comment (tomphp) [inline src/main/java/uk/gov/ons/ctp/response/action/service/impl/ActionServiceImpl.java]: How about wrapping these 2 methods into `actionPlanJob.complete(...)`?

File: src/test/java/uk/gov/ons/ctp/response/action/scheduled/PlanSchedulerIT.java [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+    assertNull(pollForPrinterAction());
+  }
+
+  private Date nowOffsetByHours(int hours) {
+    Calendar now = Calendar.getInstance();
+    now.add(Calendar.HOUR, hours);
+    return now.getTime();
+  }
+
+  private void waitForScheduler() throws InterruptedException {
```
Comment (benjefferies) [inline src/test/java/uk/gov/ons/ctp/response/action/scheduled/PlanSchedulerIT.java]: Should we fail if waitForScheduler times out?

**Comment by benjefferies** [comment]
Here's one of the intermittent failures I get
```
Failed tests:
  PlanSchedulerIT.testActiveActionPlanJobAndActionPlanCreatesAction:398 expected:<1> but was:<0>
```

File: src/main/java/uk/gov/ons/ctp/response/action/domain/repository/ActionRepository.java [hunk 3]
```diff
@@ -93,4 +95,67 @@
...
+  /**
+   * Create actions for action rules due to run
+   *
+   * @param actionPlanId action Plan primary key
+   */
+  @Modifying
+  @Query(
+      value =
+          "INSERT INTO action.action "
+              + "  ( "
```
Comment (mfcrocker) [inline src/main/java/uk/gov/ons/ctp/response/action/domain/repository/ActionRepository.java]: This formatting makes me sad

File: pom.xml [hunk 4]
```diff
@@ -223,7 +216,7 @@
         <dependency>
             <groupId>uk.gov.ons.ctp.common</groupId>
             <artifactId>test-framework</artifactId>
-            <version>10.49.10</version>
+            <version>10.49.11-SNAPSHOT</version>
```
Comment (tej99) [inline pom.xml]: Still waiting for [https://travis-ci.org/ONSdigital/rm-common-test-framework/builds/406169220?utm_source=github_status&utm_medium=notification](this)

Comment (PurpleBooth) [reply pom.xml]: https://github.com/ONSdigital/rm-common-test-framework/commit/5f67008b5758013f89fccaebc8958853b328cce0

**Pull request merged**